src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,157 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> headNode;
+    private Node<T> tailNode;
+
+    public MyLinkedList() {
+        this.size = 0;
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        addLastNode(value);
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            addLastNode(value);
+            return;
+        }
+        isIndexInBound(index);
+        addBefore(value, findNode(index));
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (T valuesFromList : list) {
+            add(valuesFromList);
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        isIndexInBound(index);
+        return findNode(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        isIndexInBound(index);
+        Node<T> founded = findNode(index);
+        T oldValue = founded.item;
+        founded.item = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        isIndexInBound(index);
+        Node<T> removed = findNode(index);
+        T value = removed.item;
+        Node<T> previousNode = removed.previous;
+        Node<T> nextNode = removed.nextNode;
+        if (nextNode == null) {
+            tailNode = previousNode;
+            removed.previous = null;
+        } else if (previousNode == null) {
+            headNode = nextNode;
+            removed.nextNode = null;
+        } else {
+            previousNode.nextNode = nextNode;
+            nextNode.previous = previousNode;
+        }
+        size--;
+        return value;
```

Comment:
let's move this logic to the separate method and use this method in the methods: remove by index and remove by value